### ISSUE TITLE
Importing only updated packages?

### ISSUE THREAD

**Issue by mazod** [aptly-dev/aptly#424]
I have two different network in the company. One has internet connection and the other doesn't  and i would like to create a debian repository for that offline/internal company network. So, i create a mirror on the online side and a empty repo on the offline side. Then i copy the content of .aptly folder from online side to the offline side. Now it's working, but do i have to copy the all packages after every update?

What i want is to copy only the updated packages on a regular basis after updates. Is it possible with snapshots? I created a snapshot of mirror, but isn't it just a list of references to packages and how am i going to move it? Is it possible to with filter options?

**Comment by kumy** [comment]
Why don't you use `rsync`? :)

**Comment by mazod** [comment]
these are two different networks actually, but I have a one direction pipeline to the offline network. So i can't use rsync.

**Comment by kumy** [comment]
Are you sure, one of this syntax should works in your case, no?

`$ rsync source host:destination`
or
`$ rsync host:source destination`

**Comment by mazod** [comment]
No. I think I should have explained my topology in a much clear way. There are two different networks. One online and the other one is offline. These two doesn't have direct connection. There is a one direction pipeline from online to offline working on a single machine. Both two networks have added that single machine as samba share for file sharing. I think it might be only possible in your solution if i copy the repo content to the pipeline machine persistently then rsync to that directory from both sides, right?

**Comment by dankegel** [comment]
I have the very same problem.  I use something like

aptly mirror update   # pull in a subset
aptly repo import      # grab from mirror
aptly repo remove --old (see https://github.com/smira/aptly/issues/291 )
aptly publish update # export
rsync to public server over a slow link

and while the aptly mirror update is nice and incremental, and the later steps do the right thing,
they also change the timestamp of the packages, which makes rsync send them all.
I think I can use --checksum on rsync to avoid this, but then I spend many minutes checksumming
gigabytes of packages.

It'd be nice if aptly repo import didn't touch packages that didn't, um, need touching.  Likewise for publish update.

(I only dimly grok what's going on, so there may be an error in the above.)

**Comment by flotho** [comment]
Hi @mazod @dankegel , did you try with the latest code ?
